### ISSUE TITLE
Accessibility Issues in FileList component

### ISSUE THREAD

**Issue by marcus-herrmann** [owncloud/phoenix#2332]
### Steps to reproduce
1. Use Phoenix with assistive technology, such as: Screen Reader

### Expected behaviour
The user should know of the existence of every interactive element and their functionsthe contents of every table column by label table headers.

### Actual behaviour
Most of the checkboxes are unlabeled, the files or folder controls are semantically the wrong element (covered in #2317), some table headers appear empty:

![Screenshot of Phoenix's interface, with added numbers, see below](https://user-images.githubusercontent.com/505181/67683793-4b935900-f992-11e9-8558-e1d4e5ff7f65.png)

1. This table header item/column header conains just an unlabel checkbox, therefore no text information that could serve as label
2. This table header item/column header is completely empty
3. This checkbox is not labelled
4. This star image is semantically speaking not a control but a clickable div (covered in ODS PR https://github.com/owncloud/owncloud-design-system/pull/525)
5. This File/Folder label is also just a clickable span (covered in #2317)

(The rest of the issue template is irrelevant to the kind of this issue)
